src/netbase.cpp

```diff
@@ -1291,6 +1291,24 @@ CSubNet::CSubNet(const std::string &strSubnet, bool fAllowLookup)
         network.ip[x] &= netmask[x];
 }
 
+CSubNet::CSubNet(const CNetAddr &addr, bool fAllowLookup)
+{
+    valid = addr.IsValid();
+    if (!valid)
+        return;
+
+    //set all netmask bits to 1
+    memset(netmask, 255, sizeof(netmask));
+
+    std::vector<CNetAddr> vIP;
+    if (!LookupHost(addr.ToString().c_str(), vIP, 1, fAllowLookup))
```

Comment:
Right. Wasn't aware of the `LookupHost()` in the `CNetAddr::CNetAddr()` constructor.
Fixed and pushed.